envoy/stats/stats.h

```diff
@@ -151,6 +151,15 @@ class Gauge : public Metric {
   virtual void inc() PURE;
   virtual void set(uint64_t value) PURE;
   virtual void sub(uint64_t amount) PURE;
+  // When adjusting a gauge both ways (e.g. buffer size update when simultaneously
+  // streaming in and out), using this helper avoids touching the atomic twice.
+  inline void adjust(uint64_t add_amount, uint64_t sub_amount) {
```

Comment:
I'm not sure about putting an inline helper in a pure interface. This is why I thought it was simpler to put this in the stats utils.

Reply:
My main rationale for preferring it here was that this is more discoverable. I would have seen it here, and would not have seen it in Util, while writing the code that benefits from it.

Reply:
This makes sense. But there's already a lot of important Stats util functions in Util (e.g. counterFromElements) and then this would be the only that wasn't.

Reply:
I'm fine with it being on the interface, since it is a simple implementation in terms of other functions on the interface.